File: typescript/behavioral/chain_of_responsibility/guru_example/squirrel_handler.ts
import {AbstractHandler} from "./abstract_handler";

export class SquirrelHandler extends AbstractHandler {
  public handle(request: string): string | null {
    if (request === 'Nut') {
      return `Squirrel: I'll eat the ${request}`
    }
    return super.handle(request);
  }
}

File: dart/behavioral/chain_of_responsibility/guru_example/bin/squirrel_handler.dart
import 'abstract_handler.dart';

class SquirrelHandler extends AbstractHandler {
  @override
  String handle(String request) {
    if (request == 'Nut') {
      return 'Squirrel: I\'ll eat the $request';
    }
    return super.handle(request);
  }
}
